presto-main/src/main/java/com/facebook/presto/operator/aggregation/StatisticalTDigest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation;
+
+import com.facebook.presto.tdigest.TDigest;
+import io.airlift.slice.Slice;
+
+public class StatisticalTDigest
+        implements StatisticalDigest<TDigest>
+{
+    private final TDigest tDigest;
+
+    public StatisticalTDigest(TDigest tDigest)
+    {
+        this.tDigest = tDigest;
+    }
+
+    @Override
+    public void add(Number value, long weight)
```

Comment:
As discussed, this might have some overhead from autoboxing.  Instead, we could just make these implementations inner classes and expose them via factory methods on the interface.